graylog2-server/src/main/java/org/graylog2/rest/resources/system/indexer/IndexSetsResource.java

```diff
@@ -0,0 +1,158 @@
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.rest.resources.system.indexer;
+
+import com.codahale.metrics.annotation.Timed;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.apache.shiro.authz.annotation.RequiresAuthentication;
+import org.apache.shiro.authz.annotation.RequiresPermissions;
+import org.graylog2.audit.AuditEventTypes;
+import org.graylog2.audit.jersey.AuditEvent;
+import org.graylog2.indexer.indexset.IndexSetConfig;
+import org.graylog2.indexer.indexset.IndexSetService;
+import org.graylog2.rest.resources.system.indexer.responses.IndexSetResponse;
+import org.graylog2.rest.resources.system.indexer.responses.IndexSetSummary;
+import org.graylog2.shared.rest.resources.RestResource;
+import org.graylog2.shared.security.RestPermissions;
+
+import javax.inject.Inject;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.ClientErrorException;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.HashSet;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+@RequiresAuthentication
+@Api(value = "System/IndexSets", description = "Index sets")
+@Path("/system/indices/index_sets")
+@Produces(MediaType.APPLICATION_JSON)
+public class IndexSetsResource extends RestResource {
+    private final IndexSetService indexSetService;
+
+    @Inject
+    public IndexSetsResource(IndexSetService indexSetService) {
+        this.indexSetService = requireNonNull(indexSetService);
+    }
+
+    @GET
+    @Timed
+    @ApiOperation(value = "Get a list of all index sets")
+    @ApiResponses(value = {
+            @ApiResponse(code = 403, message = "Unauthorized"),
+    })
+    public IndexSetResponse list() {
+        final Set<IndexSetConfig> all = indexSetService.findAll();
+        final Set<IndexSetSummary> indexSets = new HashSet<>(all.size());
+        for (IndexSetConfig indexSetConfig : all) {
```

Comment:
Heh, no particular reason. It seems I was just in "zombie" mode. 😉 

I'll change it for consistency.